### PR TITLE
fix quality flaws

### PR DESCRIPTION
**Pull request by Bertk**
- different license headers
- method needlessly boxes a boolean constant
- concatenates the result of a toString()
- passes constant String of length 1 to character overridden method 
- string literal on the left side of this string comparison
- Assertions should be complete

### CODE REVIEW

**Comment by guwirth** [comment]
@Bertk looks good, thx for providing this. Would be good a second one (maybe @jmecosta) will also have a look.

**Comment by guwirth** [comment]
Will merge this because a lot of files are changed. Risk to get merge conflicts if I delay it. @jmecosta if there are problems we have to fix it later on.

File: cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java [hunk 1]
```diff
@@ -1,493 +1,496 @@
...
-    }
-    AddMacro("_MSC_VER=1900", fileElement);
-    // VS2015 RC
-    AddMacro("_MSC_FULL_VER=190022816", fileElement);
-    //_MFC_VER Defines the MFC version. For example, in Visual Studio 2013, _MFC_VER is defined as 0x0C00.
-    AddMacro("_MFC_VER=0x0C00", fileElement);
-    AddMacro("_ATL_VER=0x0C00", fileElement);
-  }
-}
+/*
```
Comment (jmecosta) [inline cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java]: Seems to be only line endig fixing. Seems ok

Comment (jmecosta) [inline cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java]: Seems to be only line endig fixing. Seems ok

Comment (jmecosta) [inline cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java]: Seems to be only line endig fixing. Seems ok

**Pull request merged**